File: programming-concepts/logical-operators/python/xor.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def main():
    tea_chosen = False
    coffee_chosen = True

    if tea_chosen ^ coffee_chosen:
        print("Your drink is being processed")
    else:
        print("Choose either tea or coffee but not both")

        
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: programming-concepts/logical-operators/visual-basic/xor.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim teaChosen As Boolean = True
        Dim coffeeChosen As Boolean = True

        If teaChosen Xor coffeeChosen Then
            Console.WriteLine("Your drink is being processed")
        Else
            Console.WriteLine("Choose either tea or coffee but not both")
        End If
    End Sub

End Module
